platform/android/src/ch/opengis/qfield/QFieldUtils.java

```diff
@@ -57,6 +58,28 @@ public static String getArchiveProjectName(InputStream in) {
         return projectName;
     }
 
+    public static boolean documentFileToFolder(DocumentFile directory,
+                                               String folder,
+                                               ContentResolver resolver) {
+        DocumentFile[] files = directory.listFiles();
+        for (DocumentFile file : files) {
+            if (file.isDirectory()) {
+                String directoryPath = folder + file.getName() + "/";
+                new File(directoryPath).mkdir();
+                documentFileToFolder(file, directoryPath, resolver);
+            } else {
+                String filePath = folder + file.getName();
+                try {
+                    InputStream input = resolver.openInputStream(file.getUri());
+                    QFieldUtils.inputStreamToFile(input, filePath);
+                } catch (Exception e) {
+                    e.printStackTrace();
```

Comment:
Is that worth a message to the user?

Reply:
@m-kuhn , yeah, there are a few alert dialogs to put in, I'm deferring to a follow up PR to cover this and other places in the code.